### PR TITLE
Retrieve bags from Wellcome storage service

### PR DESCRIPTION
**Pull request by helenst**
This will need the service to have read access to the storage buckets. I'll make an infra PR for that.

### CODE REVIEW

File: storage_service/locations/models/wellcome.py [hunk 1]
```diff
@@ -69,14 +69,42 @@ def browse(self, path):
...
+            client_secret=self.app_client_secret,
+        )
+
     def delete_path(self, delete_path):
         LOGGER.debug('Deleting %s from Wellcome storage' % delete_path)
 
     def move_to_storage_service(self, src_path, dest_path, dest_space):
         """ Moves src_path to dest_space.staging_path/dest_path. """
-        LOGGER.debug('Fetching %s from %s (%s) on Wellcome storage' % (
+        LOGGER.debug('Fetching %s on Wellcome storage to %s (space %s)' % (
```
Comment (alexwlchan) [inline storage_service/locations/models/wellcome.py]: nit:

```suggestion
        LOGGER.debug('Fetching %s on Wellcome storage to %s (space %s)', 
```

Drop the `%`-formatting, use the more efficient comma-formatting used elsewhere.

File: storage_service/locations/models/wellcome.py [hunk 2]
```diff
@@ -69,14 +69,42 @@ def browse(self, path):
...
+                # The bag is stored unzipped (i.e. as a directory tree).
+                # Download all objects in the source directory
+                s3_prefix = loc['path'].lstrip('/')
+                objects = bucket.objects.filter(Prefix=s3_prefix)
+                for objectSummary in objects:
+                    dest_file = objectSummary.key.replace(s3_prefix, dest_path, 1)
+                    self.space.create_local_directory(dest_file)
+
+                    bucket.download_file(objectSummary.key, dest_file)
+                break
```
Comment (alexwlchan) [inline storage_service/locations/models/wellcome.py:106]: Maybe worth throwing an error here if we don't provide you with an `aws-s3-ia` location?

File: storage_service/locations/tests/test_wellcome.py [hunk 3]
```diff
@@ -138,3 +141,52 @@ def set_package_to_fail(*args):
...
+    fixtures = ['base.json', 'wellcome.json']
+
+    def setUp(self):
+        self.wellcome_object = models.WellcomeStorageService.objects.get(id=1)
+
+        self._s3 = boto3.client("s3", region_name='us-east-1')
+        self.tmp_dir = tempfile.mkdtemp()
+
+    def tearDown(self):
+        shutil.rmtree(self.tmp_dir)
```
Comment (alexwlchan) [inline storage_service/locations/tests/test_wellcome.py:156]: minor: If you're using `py.test` for tests (which I think we are?), the `tmp_dir` fixture is a tad cleaner: https://doc.pytest.org/en/latest/tmpdir.html

**Review by alexwlchan — APPROVED** [review]
All looks good. A couple of inline comments, but nothing major.

Happy for you to address as many comments as you think are necessary, then merge when ready.

File: storage_service/locations/tests/test_wellcome.py (hunk 3 above)
Comment (helenst) [reply storage_service/locations/tests/test_wellcome.py:156]: We are, although pytest fixtures don't load so easily when `TestCase` is being used (which it is, for database fixtures)

**Pull request merged**